# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build vulkan-headers-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
PKG_CONFIG_PATH=/drm/lib/pkgconfig cmake -DBUILD_WSI_WAYLAND_SUPPORT=OFF -DBUILD_WSI_XCB_SUPPORT=OFF -DBUILD_WSI_XLIB_SUPPORT=OFF -DCMAKE_INSTALL_PREFIX=/drm -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/drm/lib
install build/loader/libvulkan.so.1.* $DESTDIR/drm/lib/libvulkan.so.1
ln -sf libvulkan.so.1 $DESTDIR/drm/lib/libvulkan.so
install -d $DESTDIR/drm/lib/pkgconfig
install -m 644 build/loader/vulkan.pc $DESTDIR/drm/lib/pkgconfig

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/vulkan-loader-build.pc << EOF
Name: Vulkan-Loader
Version: 1.2.169
Description: Vulkan Installable Client Driver (ICD) loader
Requires: $REQUIRES

devel=\\
/drm/lib/libvulkan.so \\
/drm/lib/pkgconfig/vulkan.pc

exec=\\
/drm/lib/libvulkan.so.1
EOF

exit
--- Vulkan-Loader-1.2.169.orig/CMakeLists.txt
+++ Vulkan-Loader-1.2.169/CMakeLists.txt
@@ -30,7 +30,7 @@
 add_definitions(-DVK_ENABLE_BETA_EXTENSIONS)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-find_package(PythonInterp 3 QUIET)
+find_package(Python3 QUIET)
 
 if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/external/googletest)
     option(BUILD_TESTS "Build Tests" ON)
@@ -192,9 +192,9 @@
 endif()
 
 # Optional codegen target
-if(PYTHONINTERP_FOUND)
+if(Python3_Interpreter_FOUND)
     add_custom_target(VulkanLoader_generated_source
-                      COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/scripts/generate_source.py
+                      COMMAND Python3::Interpreter ${PROJECT_SOURCE_DIR}/scripts/generate_source.py
                               ${VulkanRegistry_DIR} --incremental
                       )
 else()
--- Vulkan-Loader-1.2.169.orig/loader/CMakeLists.txt
+++ Vulkan-Loader-1.2.169/loader/CMakeLists.txt
@@ -261,7 +261,7 @@
                                      "${VulkanHeaders_VERSION_MAJOR}.${VulkanHeaders_VERSION_MINOR}.${VulkanHeaders_VERSION_PATCH}")
     target_link_libraries(vulkan ${CMAKE_DL_LIBS} m)
     if (NOT ANDROID)
-        target_link_libraries(vulkan pthread)
+        target_link_libraries(vulkan rt)
     endif()
     target_link_libraries(vulkan Vulkan::Headers)
     if(APPLE)
